SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+#ifndef SMALL_BETTER_LOGGER_H
+#define SMALL_BETTER_LOGGER_H
+
+// Used for writing to output stream
+#include <iostream>
+#include <fstream>
+
+// Used for formatting and creating the output string
+#include <sstream>
+#include <vector>
+#include <regex>
+
+namespace sbl
+{
+	// Logger declaration
+	class Logger;
+	using logger = Logger;
+
+	// Stream types to be used by a Logger instance
+	enum class STREAM_TYPE
+	{
+		STDOUT, STDERR, STDLOG, FILE
+	};
+	using stream_type = STREAM_TYPE;
+
+	// Logger class: The main part of the library, used to log messages to a stream (ex.: STDOUT, STDERR, file etc.)
+	class Logger
+	{
+	public:
+		// Constructors and destructors
+
+		// Creates an instance of Logger which outputs to a stream chosen from a STREAM_TYPE (except STREAM_TYPE::FILE)
+		// Defaults to STREAM_TYPE::STDOUT and no auto flush
+		Logger(STREAM_TYPE type = STREAM_TYPE::STDOUT)
+			: m_StreamType(type), m_AutoFlush(false)
+		{
+			if (m_StreamType == STREAM_TYPE::FILE) throw std::exception("Cannot make logger with STREAM_TYPE::FILE with no file provided.");
+		}
+
+		// Creates an instance of Logger which outputs to a stream chosen from a STREAM_TYPE (except STREAM_TYPE::FILE)
+		// By default auto flush is turned on
+		Logger(const char* filepath, bool autoFlush = true)
+			: m_StreamType(STREAM_TYPE::FILE), m_AutoFlush(autoFlush)
+		{
+			if (!filepath || strlen(filepath) < 1 || *filepath == 0) throw std::exception("No file path provided. Cannot create output stream.");
+
+			m_FileStream.open(filepath);
+			if (!m_FileStream.is_open()) throw std::exception("Cannot open file to output to.");
```

Comment:
```suggestion
			if (!m_FileStream.is_open()) throw std::exception("Cannot open output file." / "Cannot open file.");
```
Also, as a good practice, solving situations where you have to change string and maybe you are not sure if any other string in the project has to change because of that, I suggest using a Constants class, holding static refrences to your string resources (or macros, or whatever is the norm in cpp, just hold all you resources somewhere). 
```suggestion
			if (!m_FileStream.is_open()) throw std::exception(Constants.OpenFileExceptionMessage);
```